asserts/ifacedecls.go

```diff
@@ -366,6 +366,16 @@ func baseCompileRule(context string, rule interface{}, target rule, subrules []s
 	return nil
 }
 
+type whichConnectionConstraints int
```

Comment:
`connectionConstraintsKind` or similar?  Feels a bit like a variable name as it is.